nestkernel/conn_parameter.h

```diff
@@ -69,8 +69,8 @@ class ConnParameter
    * @param rng   random number generator pointer
    * will be ignored except for random parameters.
    */
-  virtual double value_double( index, index, librandom::RngPtr& ) const = 0;
-  virtual long_t value_int( index, index, librandom::RngPtr& ) const = 0;
+  virtual double value_double( thread, librandom::RngPtr& ) const = 0;
+  virtual long_t value_int( thread, librandom::RngPtr& ) const = 0;
```

Comment:
I agree with both of you, we planned to implement the connection parameter in an extendible fashion. I removed the sgid and tgid arguments for two reasons. First, to remove the false impression, that the required functionality using these arguments is already in place. Second, as far as I recall we introduced the arguments to allow for parameter matrices, which are now inserted as one-dimensional arrays in order to be iterable. If we decide to extend the interface to allow for parameter-generating functions which are also iterable, the arguments sgid and tgid might not be necessary. Therefore I suggest to leave the arguments out for now and re-insert them from history if required in future development. 
The argument thread was introduced to secure the parameter array of iterators when using multi-threading.